### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by rahul1995**
## Purpose
To write the test cases for the key operations of Dropbox connector

Issue Title: Introduce a Ballerina connector for Dropbox REST API
Issue link: https://github.com/ballerina-platform/ballerina-library/issues/7054

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/tests/test.bal [hunk 1]
```diff
@@ -0,0 +1,171 @@
+// This file is auto-generated by the Ballerina OpenAPI tool.
+
+// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
```
Comment (Nuvindu) [inline ballerina/tests/test.bal]:
```suggestion
// Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
```

File: ballerina/tests/test.bal [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+configurable boolean isLiveServer = os:getEnv("IS_LIVE_SERVER") == "true";
+configurable http:BearerTokenConfig & readonly authConfig = {
+    token: isLiveServer ? os:getEnv("DROPBOX_TOKEN") : "test"
+};
+ConnectionConfig config = {auth: authConfig};
+final Client apiClient = check new Client(config, serviceUrl = isLiveServer ? "https://api.dropboxapi.com/2" : "http://localhost:9090");
+final Client contentClient = check new Client(config, serviceUrl = isLiveServer ? "https://content.dropboxapi.com/2" : "http://localhost:9090");
+
+@test:Config {}
```
Comment (Nuvindu) [inline ballerina/tests/test.bal]:
```suggestion
@test:Config {
    groups: ["live_tests", "mock_tests"]
}
```

Comment (Nuvindu) [reply ballerina/tests/test.bal]: Shall we add these test groups to all the test cases?

File: ballerina/tests/test.bal [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+import ballerina/os;
+import ballerina/test;
+
+configurable boolean isLiveServer = os:getEnv("IS_LIVE_SERVER") == "true";
+configurable http:BearerTokenConfig & readonly authConfig = {
+    token: isLiveServer ? os:getEnv("DROPBOX_TOKEN") : "test"
+};
+ConnectionConfig config = {auth: authConfig};
+final Client apiClient = check new Client(config, serviceUrl = isLiveServer ? "https://api.dropboxapi.com/2" : "http://localhost:9090");
+final Client contentClient = check new Client(config, serviceUrl = isLiveServer ? "https://content.dropboxapi.com/2" : "http://localhost:9090");
```
Comment (Nuvindu) [inline ballerina/tests/test.bal:27]: Any reason for using two seperate client instances?

File: ballerina/tests/test.bal (hunk 2 above)
Comment (rahul1995) [reply ballerina/tests/test.bal]: Done

File: ballerina/tests/test.bal (hunk 3 above)
Comment (rahul1995) [reply ballerina/tests/test.bal:27]: Dropbox categorizes endpoints in two categories: RPC and Content. All the upload/download APIs are available on server `content.dropboxapi.com` and everything else is on `api.dropboxapi.com`.

File: ballerina/tests/mock_service.bal [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+    resource function post account/set_profile_photo(@http:Payload SetProfilePhotoArg payload) returns SetProfilePhotoResult|http:Response {
+        return {
+            "profile_photo_url": "https://dl-web.dropbox.com/account_photo/get/dbaphid%3AAAHWGmIXV3sUuOmBfTz0wPsiqHUpBWvv3ZA?vers=1556069330102&size=128x128"
+        };
+    }
+
+    # create_folder
+    #
+    # + return - create_folder response 
+    resource function post files/create_folder_v2(@http:Payload CreateFolderArg payload) returns FolderMetadataResponse|http:Response {
```
Comment (lnash94) [inline ballerina/tests/mock_service.bal]:
```suggestion
    resource function post files/create_folder_v2(CreateFolderArg payload) returns FolderMetadataResponse|http:Response {
```

Comment (lnash94) [reply ballerina/tests/mock_service.bal]: Since this is `post` method we don't need to explicitly mentioned requestBody payload by adding annotation.

Comment (rahul1995) [reply ballerina/tests/mock_service.bal]: Updated all the `post` methods.

**Comment by rahul1995** [comment]
@lnash94 @Nuvindu @shafreenAnfar @NipunaRanasinghe please review.

**Comment by lnash94** [comment]
Hi @rahul1995 
If you don’t mind, could you please sync the PRs with the main branch? I noticed a few previous commits in client generation 
showing up here.

**Comment by rahul1995** [comment]
@lnash94 my bad! I have updated it now.

**Review by lnash94 — DISMISSED** [review]
LGTM

**Review by NipunaRanasinghe — COMMENTED** [review]
@rahul1995 we usually keep a README.md file inside the test directory with the instruction on how to run the tests against live server and mock server. 

You can use https://github.com/ballerina-platform/module-ballerinax-twitter/blob/main/ballerina/tests/README.md as a reference

**Comment by rahul1995** [comment]
@NipunaRanasinghe thanks for pointing that out. Added the README file.

A small suggestion: Please update the project structure in Step 1 point 4 of https://ballerina.io/learn/create-your-first-connector-with-ballerina/ so that new connector modules also follow this practice (probably the reason why I missed it too).

**Review by NipunaRanasinghe — APPROVED** [review]

**Comment by NipunaRanasinghe** [comment]
> @NipunaRanasinghe thanks for pointing that out. Added the README file.
@rahul1995 LGTM! Approved the PR 

> A small suggestion: Please update the project structure in Step 1 point 4 of https://ballerina.io/learn/create-your-first-connector-with-ballerina/ so that new connector modules also follow this practice (probably the reason why I missed it too).

Yeah your suggestion makes sense. We'll update the section with the expanded view of `tests/` directory. Thanks for pointing that out! 🙂

**Review by lnash94 — APPROVED** [review]

**Pull request merged**